src/python/pants/engine/build_files.py

```diff
@@ -212,9 +213,21 @@ def _hydrate(item_type, spec_path, **kwargs):
   return item
 
 
+class UnmatchedTargetAddressError(ResolveError):
+  @classmethod
+  def merge_error_messages(cls, errors: List[Exception]):
+    joined = '\n========= BEGIN ERROR\n'.join(str(e) for e in errors)
```

Comment:
I wish we had a consistent style guide or class to report errors, but this seems appropriately intense to me :)